app/notification_push.go

```diff
@@ -568,6 +568,47 @@ func (a *App) BuildPushNotificationMessage(contentsConfig string, post *model.Po
 	return msg, nil
 }
 
+func (a *App) SendTestPushNotification(deviceID string) (canSend bool, err error) {
+	var msg *model.PushNotification = &model.PushNotification{
+		Version:  "2",
+		Type:     model.PushTypeTest,
+		ServerId: a.TelemetryId(),
+		Badge:    -1,
+	}
+	msg.SetDeviceIdAndPlatform(deviceID)
+
+	msgJSON, jsonErr := json.Marshal(msg)
+	if jsonErr != nil {
+		return false, errors.Wrap(jsonErr, "failed to encode to JSON")
+	}
+
+	url := strings.TrimRight(*a.Config().EmailSettings.PushNotificationServer, "/") + model.APIURLSuffixV1 + "/send_push"
+	request, err := http.NewRequest("POST", url, bytes.NewReader(msgJSON))
+	if err != nil {
+		return false, err
+	}
+
+	resp, err := a.Srv().pushNotificationClient.Do(request)
+	if err != nil {
+		return false, err
+	}
+	defer resp.Body.Close()
+
+	var pushResponse model.PushResponse
+	if jsonErr := json.NewDecoder(resp.Body).Decode(&pushResponse); jsonErr != nil {
+		return false, errors.Wrap(jsonErr, "failed to decode from JSON")
+	}
+
+	switch pushResponse[model.PushStatus] {
+	case model.PushStatusRemove:
+		return false, nil
+	case model.PushStatusFail:
+		return false, errors.New(pushResponse[model.PushStatusErrorMsg])
+	}
```

Comment:
This entire part is mostly copied over from `(a *App) sendToPushProxy`. What I'd suggest is to refactor that code to handle nil session as well and name that `sendToPushProxyRaw`, and have 2 high-level methods - `sendToPushProxy` and `sendToPushProxyNoSession`, and call `sendToPushProxyNoSession` for this purpose.